### ISSUE TITLE
Custom node configuration warnings not being displayed for some nodes

### ISSUE THREAD

**Issue by arrowinaknee** [godotengine/godot#37224]
**Godot version:**
Godot 3.2.1 stable

**OS/device including version:**
Windows 10 1909

**Issue description:**
Some node classes ignore `_get_configuration_warning()` override. The problem appears with `AnimatedSpite`, `CollisionShape2D` and likely some other nodes

**Steps to reproduce:**
Create an `AnimatedSprite` node, add a script to it. Mark it as a `tool` and override the `_get_configuration_warning()` to return some message:
``` GDScript
extends AnimatedSprite
tool

func _get_configuration_warning():
    return "This message won't be displayed"
```
And indeed, only the AnimatedSprite's own warning is displayed:
![image](https://user-images.githubusercontent.com/43494664/77245211-ef292280-6c2d-11ea-8c07-08c07381a323.png)

Now let's try the same with, for example, `Bone2D` as it also displays it's own warnings:
``` GDScript
extends Bone2D
tool

func _get_configuration_warning():
    return "But this message will"
```
And here the message is displayed:
![image](https://user-images.githubusercontent.com/43494664/77245110-3e228800-6c2d-11ea-9420-74bb969b6e91.png)

I fixed the problem for the mentioned two nodes in my [fork](https://github.com/arrowinaknee/godot) but want to ensure that this is actually a bug

**Minimal reproduction project:**
https://github.com/arrowinaknee/godot_warning_issue_demo

**Comment by akien-mga** [comment]
All existing `get_configuration_warning()` overrides in core nodes need to be reviewed to ensure that they start by retrieving warnings from the parent/super class.

**Comment by Nimmda** [comment]
Hi,
I could do that by getting Parent::get_configuration_warning() and add all the warnings with an empty extra line between them like this. Do I need to assign this issue to me, when I am working on this and if, how do I do this?

```
String AnimatedSprite2D::get_configuration_warning() const {
	String warning = Node2D::get_configuration_warning();
	if (warning != String())
	{
		warning += "\n\n";
	}
	if (frames.is_null()) {
		warning += TTR("A SpriteFrames resource must be created or set in the \"Frames\" property in order for AnimatedSprite to display frames.");
	}

	return warning;
}
```

**Comment by Calinou** [comment]
> Do I need to assign this issue to me, when I am working on this and if, how do I do this?

We generally don't use GitHub's assignees feature. Even if we did, you wouldn't be able to self-assign an issue without being a member of the Godot organization anyway.

Also, there's already a pull request that addresses this: https://github.com/godotengine/godot/pull/38743

**Comment by Nimmda** [comment]
I understand. Sorry my fault and thank you for the help.

**Comment by akien-mga** [comment]
Fixed by #38743.
